### PR TITLE
Replaces basetring with str

### PR DESCRIPTION
**Pull request by tmrts**

### CODE REVIEW

**Comment by nathanielmanistaatgoogle** [comment]
This is OK to test.

**Comment by nathanielmanistaatgoogle** [comment]
See [these helpful guidelines](http://chris.beams.io/posts/git-commit/#seven-rules) about writing a good commit log message - in this case, the commit log message should say why the change is being made in addition to describing the change itself.

File: src/python/grpcio/tests/unit/framework/interfaces/links/test_utilities.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def _safe_for_log_ticket(ticket):
       possibly features values like the string "<payload of length 972321>" in
       place of the actual values of the given ticket.
   """
-  if isinstance(ticket.payload, (basestring,)):
+  if isinstance(ticket.payload, (str,)):
```
Comment (secat) [inline src/python/grpcio/tests/unit/framework/interfaces/links/test_utilities.py]: You should use the [future](https://pypi.python.org/pypi/future) which adds compatibility between Python 2.7 and Python 3.

You should put on top of the file:
```
from builtins import str
```

Comment (soltanmm) [inline src/python/grpcio/tests/unit/framework/interfaces/links/test_utilities.py]: We've already decided to use `six`, and as best I can tell the `builtins` module from `future` would require breakage from our current style guide to provide ergonomic gains. That said...

This is an interesting suggestion as far as choice-of-dependencies go. Not immediately desirable or even desirable in the long run (I'd rather cythonize in the long run), but mid-term it *might* be worth considering.

**Comment by nathanielmanistaatgoogle** [comment]
Closing; this is superseded by [pull request 6503](https://github.com/grpc/grpc/pull/6503).

**Pull request closed without merging**